src/gui/mzroll/numeric_treewidgetitem.cpp

```diff
@@ -1,15 +1,37 @@
 #include "common/alphanum.hpp"
 
 #include "numeric_treewidgetitem.h"
+#include "PeakGroup.h"
+
+static PeakGroup::ClassifiedLabel labelForString(const QString& labelString)
+{
+    if (labelString == "PeakGroup::ClassifiedLabel::Signal") {
+        return PeakGroup::ClassifiedLabel::Signal;
+    } else if (labelString == "PeakGroup::ClassifiedLabel::Noise") {
+        return PeakGroup::ClassifiedLabel::Noise;
+    } else if (labelString == "PeakGroup::ClassifiedLabel::Correlation") {
+        return PeakGroup::ClassifiedLabel::Correlation;
+    } else if (labelString == "PeakGroup::ClassifiedLabel::Pattern") {
+        return PeakGroup::ClassifiedLabel::Pattern;
+    } else if (labelString == "PeakGroup::ClassifiedLabel::CorrelationAndPattern") {
+        return PeakGroup::ClassifiedLabel::CorrelationAndPattern;
+    }
+    return PeakGroup::ClassifiedLabel::None;
+}
 
 bool NumericTreeWidgetItem::operator<( const QTreeWidgetItem & other ) const
 {
     int sortCol = treeWidget()->sortColumn();
 
-    // hack to prevent sorting of child tree-widget items (such as isotopes that
-    // need to have a fixed sort order
-    if (parent() != nullptr && sortCol != 1)
-        return false;
```

Comment:
Seems like another rebase issue. Please include this red diff too, it is important for sorting isotope/adduct child rows. It should be just below line no. 7 and above the newly added logic for sorting based on PeakML labels.